### ISSUE TITLE
IPv6 before IPv4 - IP filter

### ISSUE THREAD

**Issue by Th3Van** [storj/storj#3127]
Hey Team,

Shouldn't this be the other way around ?

```https://github.com/storj/storj/blob/5168fdfc846a99350f137a751eddd955387000dd/pkg/overlay/cache.go#L425-L434```

First filter by IPv6 then IPv4 addresses.

I'm currently running a node with both IPv6 and IPv4 (dual-stack) addresses, but the satelitte are still filtering my node by the IPv4 address.
Only way to be filtered properly (if using IPv6) is to use **only** IPv6 on the node, but Stefans satellite 118 are currently the only one using both IPv4 and IPv6. The other satellites are only using IPv4, and will therefor disqualify/pause my node. This could be resolved if the filtering was the other way around.

//Th3Van

**Comment by stefanbenten** [comment]
@Th3Van We on purpose go via IPv4 first to circumvent that issue.
As you correctly stated, the 3 other satellites do not speak IPv6 at all currently and probably do not in the foreseeable future. To circumvent DualStack users to get DQed we use the IPv4, because that is supported across the board.
In terms of IPv6 only nodes, there is no way besides just waiting for proper support, as they are properly entered with their IPv6 address in the database.

Long story short, preferring the IPv4 would create the DQ status but not as it is currently.
Please reopen, if you see things otherwise happening.
